modules/auxiliary/gather/enum_dns.rb

```diff
@@ -1,483 +1,422 @@
+      next unless r.class == Net::DNS::RR::SOA
+      records << r.mname
+      report_host(host: r.mname, info: 'SOA')
+      vprint_good("#{domain}: SOA: #{r.mname}")
     end
+    return if records.none?
+    save_loot('ENUM_SOA', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
+
+  def get_txt(domain)
+    resp = dns_query(domain, 'TXT')
+    return if resp.blank? || resp.answer.blank?
 
-    if(datastore['ENUM_TLD'])
-      tldexpnd(datastore['DOMAIN'],datastore['NS'])
+    records = []
+    resp.answer.each do |r|
+      next unless r.class == Net::DNS::RR::TXT
+      records << r.txt
+      vprint_good("#{domain}: TXT: #{r.txt}")
     end
+    return if records.none?
+    save_loot('ENUM_TXT', 'text/plain', domain, "#{records.join(',')}", domain)
+    records
+  end
 
-    if(datastore['ENUM_BRT'])
-      if wldcrd and datastore['STOP_WLDCRD']
-        print_error("Wildcard record found!")
-      else
-        dnsbrute(datastore['DOMAIN'],datastore['WORDLIST'],datastore['NS'])
-      end
+  def get_tld(domain)
+    domain_ = domain.split('.')
+    domain_.pop
+    domain_ = domain_.join('.')
+
+    tlds = [
+      'com', 'org', 'net', 'edu', 'mil', 'gov', 'uk', 'af', 'al', 'dz',
+      'as', 'ad', 'ao', 'ai', 'aq', 'ag', 'ar', 'am', 'aw', 'ac', 'au',
+      'at', 'az', 'bs', 'bh', 'bd', 'bb', 'by', 'be', 'bz', 'bj', 'bm',
+      'bt', 'bo', 'ba', 'bw', 'bv', 'br', 'io', 'bn', 'bg', 'bf', 'bi',
+      'kh', 'cm', 'ca', 'cv', 'ky', 'cf', 'td', 'cl', 'cn', 'cx', 'cc',
+      'co', 'km', 'cd', 'cg', 'ck', 'cr', 'ci', 'hr', 'cu', 'cy', 'cz',
+      'dk', 'dj', 'dm', 'do', 'tp', 'ec', 'eg', 'sv', 'gq', 'er', 'ee',
+      'et', 'fk', 'fo', 'fj', 'fi', 'fr', 'gf', 'pf', 'tf', 'ga', 'gm',
+      'ge', 'de', 'gh', 'gi', 'gr', 'gl', 'gd', 'gp', 'gu', 'gt', 'gg',
+      'gn', 'gw', 'gy', 'ht', 'hm', 'va', 'hn', 'hk', 'hu', 'is', 'in',
+      'id', 'ir', 'iq', 'ie', 'im', 'il', 'it', 'jm', 'jp', 'je', 'jo',
+      'kz', 'ke', 'ki', 'kp', 'kr', 'kw', 'kg', 'la', 'lv', 'lb', 'ls',
+      'lr', 'ly', 'li', 'lt', 'lu', 'mo', 'mk', 'mg', 'mw', 'my', 'mv',
+      'ml', 'mt', 'mh', 'mq', 'mr', 'mu', 'yt', 'mx', 'fm', 'md', 'mc',
+      'mn', 'ms', 'ma', 'mz', 'mm', 'na', 'nr', 'np', 'nl', 'an', 'nc',
+      'nz', 'ni', 'ne', 'ng', 'nu', 'nf', 'mp', 'no', 'om', 'pk', 'pw',
+      'pa', 'pg', 'py', 'pe', 'ph', 'pn', 'pl', 'pt', 'pr', 'qa', 're',
+      'ro', 'ru', 'rw', 'kn', 'lc', 'vc', 'ws', 'sm', 'st', 'sa', 'sn',
+      'sc', 'sl', 'sg', 'sk', 'si', 'sb', 'so', 'za', 'gz', 'es', 'lk',
+      'sh', 'pm', 'sd', 'sr', 'sj', 'sz', 'se', 'ch', 'sy', 'tw', 'tj',
+      'tz', 'th', 'tg', 'tk', 'to', 'tt', 'tn', 'tr', 'tm', 'tc', 'tv',
+      'ug', 'ua', 'ae', 'gb', 'us', 'um', 'uy', 'uz', 'vu', 've', 'vn',
+      'vg', 'vi', 'wf', 'eh', 'ye', 'yu', 'za', 'zr', 'zm', 'zw', 'int',
+      'gs', 'info', 'biz', 'su', 'name', 'coop', 'aero']
+
+    records = []
+    tlds.each do |tld|
+      tldr = get_a("#{domain_}.#{tld}")
```

Comment:
Thanks -- I'm not thrilled with having to rescue `ArgumentError`, but that is the only option you have right now. I've filed https://github.com/rapid7/metasploit-framework/issues/6447 to get that fixed.